sources/Inputs/src/Configuration.cpp

```diff
@@ -41,6 +42,28 @@ updateValue(T& value, const boost::property_tree::ptree& tree, const std::string
     value = value_opt->get_value<T>();
   }
 }
+
+/**
+ * @brief Helper function to update the active power compensation parameter
+ *
+ * @param activePowerCompensation the value to update
+ * @param tree the element of the boost tree
+ */
+static void
+updateActivePowerCompensationValue(Configuration::ActivePowerCompensation& activePowerCompensation, const boost::property_tree::ptree& tree) {
+  std::map<std::string, Configuration::ActivePowerCompensation> enumResolver = {{"P", Configuration::ActivePowerCompensation::P},
+                                                                                {"targetP", Configuration::ActivePowerCompensation::TARGET_P},
+                                                                                {"PMax", Configuration::ActivePowerCompensation::PMAX}};
+  std::string apcString;
+  helper::updateValue(apcString, tree, "ActivePowerCompensation");
+  auto it = enumResolver.find(apcString);
+  if (it != enumResolver.end()) {
+    activePowerCompensation = it->second;
+  } else if (!apcString.empty()) {
+    //LOG(warn) << MESS(BadActivePowerCompensation) << LOG_ENDL; //This doesn't work, I think it always expect args
+    LOG(warn) << "The ActivePowerCompensation parameter was wrongly entered in the config file, the different possibility are : PMAX, P or targetP. The default value of PMAX will be used." << LOG_ENDL;
```

Comment:
What if this option is not in the config and that we want to use the default value?

Reply:
This is what the test !apcString.empty() check. If the option is not in the config the string won't be updated by helper::updateValue.
It may not be very clear, if you want me to change it or add a comment, please say so.

Reply:
helper::updateValue does not update the value if it doesn't exist, resulting in an empty string if the config is not given. Since the default configuration is set as a default value for the member, it will be OK.